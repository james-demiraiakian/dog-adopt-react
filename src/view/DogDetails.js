import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import Dog from '../components/Dog/Dog';
import { getDogById } from '../services/dogs';

export default function DogDetails() {
  const [dog, setDog] = useState([]);
  const [loading, setLoading] = useState(true);
  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const data = await getDogById(params.id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [params.id]);

  if (loading) return <h1>LOADING</h1>;

  return <Dog {...dog} />;
}
