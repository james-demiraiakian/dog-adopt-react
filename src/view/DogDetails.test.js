import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogDetails from './DogDetails';

test('should render a dog', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/4']}>
      <Route path="/dogs/:id" component={DogDetails} />
    </MemoryRouter>
  );

  await screen.findByText('Molly');

  expect(container).toMatchSnapshot();
});
