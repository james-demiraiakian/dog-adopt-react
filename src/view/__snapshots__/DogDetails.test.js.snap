// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a dog 1`] = `
<div>
  <div
    class="dog-detail"
  >
    <h2>
      Molly
    </h2>
    <img
      class="dog-detail-image"
      src="https://generator.ninja/static/50a86cb17862a23623a7cac32d0db051/dc4db/appenzeller-dog.jpg"
    />
    <h4>
      Age: 
      4
    </h4>
    <h4>
      Breed: 
      Appenzeller
    </h4>
    <h4>
      Bio: 
      Eye see you.
    </h4>
    <a
      href="/dogs/4/edit"
    >
      Edit
    </a>
  </div>
</div>
`;
